client/me/purchases/payment/edit-card-details/index.jsx

```diff
@@ -86,6 +92,10 @@ const EditCardDetails = React.createClass( {
 		this.setState( { form: this.formStateController.getInitialState() } );
 	},
 
+	componentWillReceiveProps( nextProps ) {
+		this.redirectIfDataIsInvalid( nextProps );
```

Comment:
Not related to this PR at all, but do you think we should add a call to `recordPageView` here, like we do in `client/me/purchases/cancel-private-registration/index.jsx`?